main.go

```diff
@@ -232,6 +232,15 @@ func (h k8sHelpers) GetNode(cli *client.Client) (*api.Node, error) {
 }
 
 func (h k8sHelpers) AddLabels(n *api.Node, labels Labels) {
+	// Remove nfd node labels that no longer apply
```

Comment:
Yes. I think it is better if we split it into another function. I would prefer calling it `RemoveLabels` similar to `AddLabels`.